ui/src/adapters/api/issuer-state.ts

```diff
@@ -26,7 +26,7 @@ const transactionStatusParser = getStrictParser<TransactionStatus>()(
 const transactionParser = getStrictParser<Transaction>()(
   z.object({
     id: z.number(),
-    publishDate: z.coerce.date(),
+    publishDate: z.coerce.date(z.string().datetime()),
```

Comment:
Why do we get an error if we just do `z.string().datetime()` without the `coerce`? I don't feel courageous enough to `infer` the cathedral ...

Reply:
Because `z.string().datetime()` first parses a string and then makes sure it has the expected datetime format, leaving the value untouched as a string.

Our type though dictates that `publishDate` must be a `Date`, so after we make sure the string is a valid datetime, we have to coerce it (convert it) to a `Date`.